src/containers/ProfileContainer/ProfileContainer.js

```diff
@@ -14,9 +20,49 @@ import style from './style.scss';
 class ProfileContainer extends Component {
   constructor(props) {
     super(props);
+    const userInfo = this.props.userInfo || {};
+    this.state = {
+      profileDetails: {
+        githubLogin: userInfo.githubLogin,
+        firstName: userInfo.firstName,
+        lastName: userInfo.lastName,
+        email: userInfo.email,
+        phoneNumber: userInfo.phoneNumber,
+        country: userInfo.country
+      }
+    };
+  }
+
+  componentWillReceiveProps(props) {
+    if (!this.props.userInfo && props.userInfo) {
+      const newProfileDetails = _.pick(props.userInfo, Object.keys(this.state.profileDetails));
+      this.setState({profileDetails: newProfileDetails});
+    }
+  }
+
+  onChangeProfileDetail = (title, value) => {
+    const newProfileDetails = {...this.state.profileDetails};
+    newProfileDetails[title] = value;
+    this.setState({profileDetails: newProfileDetails});
+  }
+
+  onResetProfileDetails = (e) => {
+    e.preventDefault();
+    this.setState({userInfo: _.pick(this.props.userInfo, Object.keys(this.state.profileDetails))});
+  }
+
+  onSubmitProfileDetails = (e) => {
+    e.preventDefault();
+    this.props.userEdit({id: this.props.userInfo._id, ...this.state.profileDetails});
   }
 
   render() {
+    const {userInfo} = this.props;
+    if (!userInfo) {
```

Comment:
Please move down this part and put it in other hook

Reply:
Done